hoomd/md/CMakeLists.txt

```diff
@@ -274,7 +274,7 @@ else()
     target_link_libraries(_md PUBLIC _hoomd)
 endif()
 if (ENABLE_CUDA)
-    target_link_libraries(_md PUBLIC _neighbor)
+    target_link_libraries(_md PRIVATE neighbor::neighbor)
```

Comment:
Something feels funny about this, but I couldn't get CMake to run properly without using `PRIVATE`.

Reply:
With public `PUBLIC` cmake will add the neighbor include directories to any target that links `_md`. This is needed if `neighbor` headers are included in any of md's header files. 

With `PRIVATE`, it only adds the neighbor include directories to `_md`. This is fine if `neighbor` headers are only included in md's .cc files.

Reply:
OK, thanks! That makes sense. I originally used `PUBLIC` because I did have some of the neighbor headers in `NeighborListGPUTree.cuh`. But, then I got a CMake error about not having the neighbor target in the export set for hoomd, so I switched to `PRIVATE`--this is wrong for that setup.

I think it makes alot of sense to hide the internals of `neighbor` from the public API--users just want the neighbor list objects and not the underlying implementation. I refactored so that `neighbor` is now only included in `NeighborListGPUTree.cu`, which means that `PRIVATE` should be correct.